home_page/implications/script.js

```diff
@@ -311,56 +311,104 @@ function renderEquationList(sortBy = 'index', sortOrder = 'asc') {
 
 
 function renderImplications(index) {
-    updateUrl(index+1);
-
-    if (index === null || index < 0 || index >= equations.length) {
+    if (index === null || index === undefined) {
         console.error('Invalid equation index:', index);
         return;
     }
 
-    currentEquationIndex = index;
-    selectedEquation.textContent = equations[index];
-    selectedEquation.dataset.index = index;
+    let bigIndex;
+    try {
+        bigIndex = toBigIntSafe(index);
+    } catch (err) {
+        console.error('Unable to coerce equation index to BigInt:', index, err);
+        return;
+    }
 
-    if (commentary[index+1] === undefined) {
-        hideVisibility("equationCommentary")
-        equationCommentary.innerHTML = "";
+    if (bigIndex < 0n) {
+        console.error('Invalid equation index (negative):', bigIndex.toString());
+        return;
+    }
+
+    const eqId = bigIndex + 1n;
+    const eq = Equation.fromId(eqId);
+    updateUrl(eqId);
+
+    currentEquationIndex = bigIndex;
+    selectedEquation.textContent = `Equation${eqId.toString()}[${eq.toString()}]`;
+    selectedEquation.dataset.index = bigIndex.toString();
+
+    const dualEq = eq.dual();
+    const dualIndex = dualEq.id;
+    if (dualIndex !== null) {
+        const dualDisplay = (dualIndex - 1n).toString();
+        selectedEquationDual.innerHTML = `(Dual equation: <a class='link' onclick="renderImplications('${dualDisplay}')">Equation${dualIndex.toString()}[${dualEq}]</a>)`;
     } else {
-        showVisibility("equationCommentary")
-        equationCommentary.innerHTML = commentary[index+1];
+        selectedEquationDual.innerHTML = "";
     }
 
+    const equationsArray = (typeof equations !== 'undefined' && Array.isArray(equations)) ? equations : null;
```

Comment:
Not sure why we need this for equations/commentary, if they're not defined it seems fine to just crash. Unless I'm missing some case when the explorer runs and they're actually legitimately not defined?